packages/snaps-simulator/src/features/renderer/components/Box.tsx

```diff
@@ -0,0 +1,28 @@
+import { Box as ChakraBox } from '@chakra-ui/react';
+import type { JSXElement } from '@metamask/snaps-sdk/jsx';
+import { assertJSXElement } from '@metamask/snaps-sdk/jsx';
+import { getJsxChildren } from '@metamask/snaps-utils';
+import { assert } from '@metamask/utils';
+import type { FunctionComponent } from 'react';
+
+import { SnapComponent } from '../SnapComponent';
+
+export type BoxProps = {
+  id: string;
+  node: unknown;
+};
+
+export const Box: FunctionComponent<BoxProps> = ({ node, id }) => {
+  assertJSXElement(node);
+  assert(node.type === 'Box', 'Expected value to be a Box component.');
+  return (
+    <ChakraBox key={id}>
+      {getJsxChildren(node).map((child, index) => (
+        <SnapComponent
+          node={child as JSXElement}
```

Comment:
This is not necessarily a `JSXElement`. It could also be `null` or `string` IIRC.

Reply:
We'll never have a string passed here, `getJsxChildren` return type has `string` but the only time a string is a children is with a `Text` component that has another method to render its children